src/main/java/io/leangen/graphql/util/Scalars.java

```diff
@@ -174,8 +174,8 @@ public Object parseValue(Object input) {
 
             @Override
             public Object parseLiteral(Object input) {
-                if (!(input instanceof ObjectValue)) return null;
-                return parseFieldValue(((ObjectValue) input));
+                if ((input instanceof ObjectValue)) return null;
```

Comment:
This is very fishy... Why would it return `null` if the input is an instance of `ObjectValue`? It was intended exactly to handle objects.

Further, why is it that you map a field to an _object_ scalar and not expect it to be a object?
I guess if the field could sometimes be an object and sometimes a simple value, it would make sense to map it as an object scalar. In case the whole line checking `input instanceof ObjectValue` needs to go.
But I'm afraid that would encourage bad habits, like using `Object` where a simpler types would do just fine...

What exactly is your use case? Is it in your case allowed to sometimes send a number and sometimes an object?

Reply:
It seems Scalars.parseLiteral is not done properly. See the previous code below, if the input is StringValue/IntValue/FloatValue/BooleanValue , it never reaches parseFieldValue.

```java
            @Override
            public Object parseLiteral(Object input) {
                if (!(input instanceof ObjectValue)) return null;
                return parseFieldValue(((ObjectValue) input));
            }

            private Object parseFieldValue(Value value) {
                if (value instanceof StringValue) {
                    return ((StringValue) value).getValue();
                }
                if (value instanceof IntValue) {
                    return ((IntValue) value).getValue();
                }
                if (value instanceof FloatValue) {
                    return ((FloatValue) value).getValue();
                }
                if (value instanceof BooleanValue) {
                    return ((BooleanValue) value).isValue();
                }
                if (value instanceof EnumValue) {
                    ((EnumValue) value).getName();
                }
                if (value instanceof ArrayValue) {
                    return ((ArrayValue) value).getValues().stream()
                            .map(this::parseFieldValue)
                            .collect(Collectors.toList());
                }
                if (value instanceof ObjectValue) {
                    return ((ObjectValue) value).getObjectFields().stream()
                            .collect(Collectors.toMap(ObjectField::getName, field -> parseFieldValue(field.getValue())));
                }
                //Should never happen, as it would mean the variable was not replaced by the parser
                throw new IllegalArgumentException("Unsupported scalar value type: " + value.getClass().getName());
            }
```
After reading your comment, it should be like this ?

```java
   public Object parseLiteral(Object input) {
                if (!(input instanceof Value)) return null;
                return parseFieldValue(((Value) input));
   }
```

Reply:
It is always going to be a `Value` so the `instanceof` check is not needed. But, there should be an explicit check inside the `parseFieldValue` for `NullValue`. This code os from the time before `null` existed in GraphQL. That's why it's missing